web-ui/view/gitlab.ml

```diff
@@ -25,6 +25,21 @@ let gitlab_mr_url ~org ~repo id =
 
 let format_org org = li [ a ~a:[ a_href (org_url org) ] [ txt org ] ]
 
+let ref_name r = 
+  match Astring.String.cuts ~sep:"/" r with
+  | "refs" :: "heads" :: branch -> Astring.String.concat ~sep:"/" branch
+  | [ "refs"; "pull"; id; "head" ] -> id
+  | _ -> Fmt.str "Bad ref format %S" r
+
+let ref_breadcrumb r head_hash =
+  match Astring.String.cuts ~sep:"/" r with
+  | "refs" :: "heads" :: branch ->
+      let branch = Astring.String.concat ~sep:"/" branch in
+      (branch, Fmt.str "commit/%s" head_hash)
+  | [ "refs"; "pull"; id; "head" ] ->
+      ("#" ^ id, Fmt.str "pull/%s" head_hash)
+  | _ -> (Fmt.str "Bad ref format %S" r, "")
```

Comment:
GitLab doesn't use the same format for `Merge_request`, you need to check but it's something like `refs/merge_requests/<id>/head`.

Reply:
```suggestion
   let f (u, commit_title, status, t) =
        let created_at = Timestamps_durations.pp_timestamp (Some t) in
        let short_hash = short_hash u in
        let commit_uri = commit_url ~org ~repo u in
        Build.commit_row ~commit_title ~short_hash ~created_at ~status ~commit_uri
   in List.map f
```
Isn't it a `List.map` instead of a `List.fold_left`?